File: Samples/CslaLight/vb/Rolodex/Business/Server/BusinessClasses/Ranks.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Csla
Imports Csla.Security
Imports Csla.Core
Imports Csla.Serialization
Imports Csla.Silverlight
Imports Csla.Validation

Namespace Rolodex.Business.BusinessClasses
  <Serializable, MobileFactory("Rolodex.Business.BusinessClasses.RanksFactory, Rolodex.Business", "FetchRanks")> _
  Public Class Ranks
	  Inherits NameValueListBase(Of Integer, String)
#If SILVERLIGHT Then
	Public Sub New()
	End Sub

	Public Shared Sub GetRanks(ByVal handler As EventHandler(Of DataPortalResult(Of Ranks)))
	  Dim dp As DataPortal(Of Ranks) = New DataPortal(Of Ranks)()
	  AddHandler dp.FetchCompleted, handler
	  dp.BeginFetch()
	End Sub
#End If

	Friend Sub SetReadOnlyFlag(ByVal flag As Boolean)
	  IsReadOnly = flag
	End Sub

	Public Function GetRankName(ByVal rank As Integer) As String
	  Dim returnValue As String = String.Empty
	  For Each oneItem In Me
		If oneItem.Key = rank Then
		  returnValue = oneItem.Value
		  Exit For
		End If
	  Next oneItem
	  Return returnValue
	End Function
  End Class
End Namespace

File: Samples/CslaLight/cs/Rolodex/Business/Server/BusinessClasses/Ranks.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Csla;
using Csla.Security;
using Csla.Core;
using Csla.Serialization;
using Csla.Silverlight;
using Csla.Validation;

namespace Rolodex.Business.BusinessClasses
{
  [Serializable]
  [MobileFactory("Rolodex.Business.BusinessClasses.RanksFactory, Rolodex.Business", "FetchRanks")]
  public class Ranks : NameValueListBase<int, string>
  {
#if SILVERLIGHT
    public Ranks() { }

    public static void GetRanks(EventHandler<DataPortalResult<Ranks>> handler)
    {
      DataPortal<Ranks> dp = new DataPortal<Ranks>();
      dp.FetchCompleted += handler;
      dp.BeginFetch();
    }
#endif

    internal void SetReadOnlyFlag(bool flag)
    {
      IsReadOnly = flag;
    }

    public string GetRankName(int rank)
    {
      string returnValue = string.Empty;
      foreach (var oneItem in this)
      {
        if (oneItem.Key == rank)
        {
          returnValue = oneItem.Value;
          break;
        }
      }
      return returnValue;
    }
  }
}
